changelog/deprecate_cast_from_typetuple_to_other_types.dd

```diff
@@ -0,0 +1,14 @@
+Deprecated weired casting from TypeTuple to other types
+
+The result of casting from TypeTuple to other types is puzzling
+and not useful.
+
+---
+TypeTuple!(int, int) values;
+auto values2 = cast(long)values;
+// typeof(values2) == TypeTuple!(int, int)
```

Comment:
Should that be `TypeTuple!(long, long)`?

Reply:
It looks still confusing for me.
I think `cast(TypeTuple!(long, long))` should works in this case.

Reply:
That assumes explicit knowledge of `values.length` (allthouth it could be emulated using `std.meta.Repeat`...).
Still not a good reason to break with the assumption that operations on tuples are propagated to their members.

Reply:
Hmm.. the specification says that `A CastExpression converts the UnaryExpression to Type`, so I expects `typeof(values2)` returns `long` in the above case.
https://dlang.org/spec/expression.html#cast_expressions

Or, would we change the spec: `A CastExpression usually converts the UnaryExpression to Type` if we keeps it?